src/xmipp/libraries/parallel/mpi_classify_CL2D_core_analysis.cpp

```diff
@@ -279,13 +278,14 @@ void ProgClassifyCL2DCore::computeStableCores()
                     VEC_ELEM(maximalCoocurrence,i)=VEC_ELEM(maximalCoocurrence,j)=1;
 
                 // Now compute core
-                FOR_ALL_OBJECTS_IN_METADATA(thisClass)
+                for (size_t objId : thisClass.ids())
                 {
-                    thisClass.getValue(MDL_IMAGE,fnImg,__iter.objId);
+                    thisClass.getValue(MDL_IMAGE,fnImg,objId);
                     size_t idx=thisClassOrder[fnImg];
                     if (VEC_ELEM(maximalCoocurrence,idx))
                     {
-                        thisClass.getRow(row,__iter.objId);
+                        MDRowVec row;
+                        thisClass.getRow(row, objId);
                         thisClassCore.addRow(row);
```

Comment:
can we do sth like `addRow(getRow(objId))`?

Reply:
We can! (Because I introduced new getter returning `std::unique_ptr`). Done.